Index: ../PythonProject_modul_2/data_structure_.py
===================================================================
diff --git a/../PythonProject_modul_2/data_structure_.py b/../PythonProject_modul_2/data_structure_.py
deleted file mode 100644
--- a/../PythonProject_modul_2/data_structure_.py	(revision 109730289cb691429d3a15c107bf098a18410e69)
+++ /dev/null	(revision 109730289cb691429d3a15c107bf098a18410e69)
@@ -1,24 +0,0 @@
-def data_structure_sum(*args):
-    sum = 0
-    for element in args:
-        if isinstance(element, str):
-            sum += len(element)
-        elif isinstance(element, int):
-            sum += element
-        elif isinstance(element, float):
-            sum += element
-        elif isinstance(element, bool):
-            sum += element
-        elif isinstance(element, list):
-            sum += data_structure_sum(*element)
-        elif isinstance(element, tuple):
-            sum += data_structure_sum(*element)
-        elif isinstance(element, set):
-            sum += data_structure_sum(*element)
-        elif isinstance(element, dict):
-            sum += data_structure_sum(*tuple(element.items()))
-    return sum
-
-data_structure = [[1, 2, 3],{'a': 4, 'b': 5},(6, {'cube': 7, 'drum': 8}),"Hello",((), [{(2, 'Urban', ('Urban2', 35))}])]
-result = data_structure_sum(data_structure)
-print(result)
\ No newline at end of file
